### PR TITLE
Allow passing parameters to GET requests

### PR DESCRIPTION
**Pull request by guywithnose**

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -196,22 +196,27 @@ public function index($resource, array $filters = [])
      *
      * @param string $resource
      * @param string $id
+     * @param array  $parameters
      *
      * @return mixed opaque handle to be given to endGet()
      */
-    public function startGet($resource, $id)
+    public function startGet($resource, $id, $parameters = [])
```
Comment (chadicus) [inline src/Client.php]: type hint

File: src/ClientInterface.php [hunk 2]
```diff
@@ -33,15 +33,16 @@ function index($resource, array $filters = []);
      *
      * @param string $resource
      * @param string $id
+     * @param array  $parameters
      *
      * @return mixed opaque handle to be given to endGet()
      */
-    function startGet($resource, $id);
+    function startGet($resource, $id, $parameters);
```
Comment (chadicus) [inline src/ClientInterface.php]: type hint

File: src/ClientInterface.php [hunk 3]
```diff
@@ -33,15 +33,16 @@ function index($resource, array $filters = []);
...
 
     /**
      * @see startGet()
      */
-    function get($resource, $id);
+    function get($resource, $id, $parameters);
```
Comment (chadicus) [inline src/ClientInterface.php]: type hint

File: src/Client.php [hunk 4]
```diff
@@ -196,22 +196,27 @@ public function index($resource, array $filters = [])
...
+        }
+
         return $this->_start($url, 'GET');
     }
 
     /**
      * @see startGet()
      */
-    public function get($resource, $id)
+    public function get($resource, $id, $parameters = [])
```
Comment (chadicus) [inline src/Client.php]: type hint

File: src/ClientInterface.php (hunk 3 above)
Comment (guywithnose) [inline src/ClientInterface.php]: done

File: src/Client.php [hunk 5]
```diff
@@ -196,22 +196,27 @@ public function index($resource, array $filters = [])
...
+     * @param array  $parameters
      *
      * @return mixed opaque handle to be given to endGet()
      */
-    public function startGet($resource, $id)
+    public function startGet($resource, $id, array $parameters = [])
     {
         Util::throwIfNotType(['string' => [$resource, $id]], true);
         $url = "{$this->_baseUrl}/" . urlencode($resource) . '/' . urlencode($id);
+        if (!empty($parameters)) {
```
Comment (chadicus) [inline src/Client.php]: I don't think the `if` is actually needed here. As the url would be `/vLatest/foos/1?` if `$parameters` is empty. Which is still a valid request

**Pull request merged**